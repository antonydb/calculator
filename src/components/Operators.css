.operators {
  display: grid;
  grid-auto-rows: 1fr;
  grid-gap: 10px;
  grid-template-columns: 1fr;
}
