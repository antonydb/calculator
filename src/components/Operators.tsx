import * as React from "react";
import { Button, ButtonThemes } from "./Button";
import "./Operators.css";

export enum Symbols {
  Multiply = "ร",
  Divide = "รท",
  Subtract = "-",
  Add = "+",
}

export enum Operations {
  Multiply = "*",
  Divide = "/",
  Subtract = "-",
  Add = "+",
}

const OperationsDict = {
  [Operations.Multiply]: Symbols.Multiply,
  [Operations.Divide]: Symbols.Divide,
  [Operations.Subtract]: Symbols.Subtract,
  [Operations.Add]: Symbols.Add,
};

interface OperatorsProps {
  onClick: (val: string) => void;
}

export const Operators: React.FC<OperatorsProps> = ({ onClick }) => (
  <div className="operators">
    {Object.entries(OperationsDict).map(([value, text]) => (
      <Button
        key={value}
        theme={ButtonThemes.Primary}
        value={value}
        onClick={onClick}
      >
        {text}
      </Button>
    ))}
    <Button theme={ButtonThemes.Primary} type="submit">
      =
    </Button>
  </div>
);
