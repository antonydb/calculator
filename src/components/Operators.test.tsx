import * as React from "react";
import { render } from "@testing-library/react";
import { Operators, Symbols } from "./Operators";

describe("<Operators />", () => {
  const { container } = render(<Operators onClick={() => undefined} />);

  it("should match snapshot", () => {
    expect(container.innerHTML).toMatchSnapshot();
  });
});
