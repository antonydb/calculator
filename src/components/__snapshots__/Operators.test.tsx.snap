// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Operators /> should match snapshot 1`] = `"<div class=\\"operators\\"><button class=\\"button button-primary\\" type=\\"button\\" value=\\"*\\">ร</button><button class=\\"button button-primary\\" type=\\"button\\" value=\\"/\\">รท</button><button class=\\"button button-primary\\" type=\\"button\\" value=\\"-\\">-</button><button class=\\"button button-primary\\" type=\\"button\\" value=\\"+\\">+</button><button class=\\"button button-primary\\" type=\\"submit\\">=</button></div>"`;
